src/Microsoft.SqlTools.Dmp.Hosting/Extensibility/ExtensionServiceProvider.cs

```diff
@@ -0,0 +1,143 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Composition.Convention;
+using System.Composition.Hosting;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using Microsoft.SqlTools.Dmp.Hosting.Utility;
+
+namespace Microsoft.SqlTools.Dmp.Hosting.Extensibility
+{
+    public class ExtensionServiceProvider : RegisteredServiceProvider
```

Comment:
Seems like I missed a class definition for this? 
```C#
/// <summary>
/// A MEF-based service provider. Supports any MEF-based configuration but is optimized for service discovery over a set of DLLs in an application scope. 
/// Any service registering using the [Export(IServiceContract)] attribute will be discovered and used by this service provider if it's in the set of Assemblies / Types specified during its construction. 
/// Manual override of this is supported by calling <see cref="RegisteredServiceProvider.RegisterSingleService" /> and similar methods, since this will initialize that service contract and avoid the MEF-based search and discovery process.
/// This allows the service provider to link into existing singleton / known services while using MEF-based dependency injection and inversion of control for most of the code.
/// </summary>
```